docs/reference/mapping/dynamic/field-mapping.asciidoc

```diff
@@ -1,35 +1,37 @@
 [[dynamic-field-mapping]]
 === Dynamic field mapping
 
-By default, when a previously unseen field is found in a document,
-Elasticsearch will add the new field to the type mapping.   This behaviour can
-be disabled, both at the document and at the <<object,`object`>>  level, by
-setting the <<dynamic,`dynamic`>> parameter to `false` (to ignore new fields) or to `strict` (to throw
-an exception if an unknown field is encountered).
-
-Assuming `dynamic` field mapping is enabled, some simple rules are used to
-determine which data type the field should have:
-
-[horizontal]
-*JSON data type*::                   *Elasticsearch data type*
-
-`null`::                            No field is added.
-`true` or `false`::                 <<boolean,`boolean`>> field
-floating{nbsp}point{nbsp}number::   <<number,`float`>> field
-integer::                           <<number,`long`>> field
-object::                            <<object,`object`>> field
-array::                             Depends on the first non-`null` value in the array.
-string::                            Either a <<date,`date`>> field
-                                        (if the value passes <<date-detection,date detection>>),
-                                    a <<number,`double`>> or <<number,`long`>> field
-                                        (if the value passes <<numeric-detection,numeric detection>>)
-                                    or a <<text,`text`>> field, with a <<keyword,`keyword`>> sub-field.
-
-These are the only <<mapping-types,field data types>> that are dynamically
-detected.  All other data types must be mapped explicitly.
-
-Besides the options listed below, dynamic field mapping rules can be further
-customised with <<dynamic-templates,`dynamic_templates`>>.
+When {es} detects a new field in a document, it _dynamically_ adds the field to
+the type mapping. The <<dynamic,`dynamic`>> parameter controls this behavior.
+
+You can disable dynamic mapping, both at the document and at the
+<<object,`object`>> level. Setting the `dynamic` parameter to
+`false` ignores new fields, and `strict` rejects the document if {es}
+encounters an unknown field.
+
+When dynamic field mapping is enabled, {es} uses the following rules to
```

Comment:
Is it clear to users that by "enabled" we don't only mean "true" but also "runtime". It becomes clearer with the following table but maybe that could be further clarified?

Reply:
I think we can expand to explicitly mention `runtime` here:
>Dynamic field mapping is enabled by default, but can also be enabled by setting the `dynamic` parameter to `runtime`. When dynamic field mapping is enabled, Elasticsearch uses the rules in the following table to determine how to map data types for each field.

Reply:
If I read this I would wonder if dynamic:true and dynamic:runtime do the same thing. I think we should be more explicit that dynamic:true and dynamic:runtime are both going to dynamically create fields based on incoming documents, and that both are what we refer to as dynamic field mapping, as opposed to dynamic:false and dynamic:strict which don't create fields dynamically.

Reply:
@javanna, I'm almost inclined to just pull this same table from the [`dynamic`](https://www.elastic.co/guide/en/elasticsearch/reference/master/dynamic.html#dynamic-parameters) page into this page, since it clearly explains the different settings.

Reply:
Thanks @javanna -- I'm moving this discussion to a draft PR so that we can collaborate on changes there. See #66530.